### DISCUSSION TITLE
Tvheadend and kodi with nginx proxy manager

### DISCUSSION THREAD

**Discussion by alebupal** [NginxProxyManager/nginx-proxy-manager#1558] [Q&A] [answered]
Hi, I am trying to configure tvheadend using npm to connect it with kodi.
Tvheadend by default uses port 9981 to access the web

Kodi asks me for these ports:
https://i.ibb.co/T0L0gwS/Sin-t-tulo.jpg

I have the npm configuration like this:
https://prnt.sc/1imnr22

How can I get the port 9982 that kodi asks for to work?

**Comment by chaptergy** [comment, marked as answer]
You can add a stream host for that. You will also need to add the port mapping for port 9982 to your docker-compose file.

**Reply by alebupal** [reply]
> You can add a stream host for that. You will also need to add the port mapping for port 9982 to your docker-compose file.

I have tvheadend on another device other than npm with another ip, should I indicate the ip somewhere?
In the ports of the docker-compose.yml file I think I can only put a port and not an ip with the port.
Thanks!

**Reply by chaptergy** [reply]
Go to hosts > stream > add stream and add something like this
![image](https://user-images.githubusercontent.com/26956711/140311029-f2c367b9-52e1-493f-b458-91b40b6c1a32.png)

Then edit your docker-compose file and add something like this:

```yml
version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    ports:
      - '9982:9982' # Add this line!
      [...]
```
If you chose some other port as incoming port you of course need to update the port mapping in your yml as well.

**Reply by alebupal** [reply]
> Go to hosts > stream > add stream and add something like this ![image](https://user-images.githubusercontent.com/26956711/140311029-f2c367b9-52e1-493f-b458-91b40b6c1a32.png)
> 
> Then edit your docker-compose file and add something like this:
> 
> ```yaml
> version: '3'
> services:
>   app:
>     image: 'jc21/nginx-proxy-manager:latest'
>     ports:
>       - '9982:9982' # Add this line!
>       [...]
> ```
> 
> If you chose some other port as incoming port you of course need to update the port mapping in your yml as well.

This does not work for me, I put screenshots

IP NPM: http://192.168.5.5:81
IP tvheadend: http://192.168.5.8:9981/
config host:
https://i.ibb.co/yYPQ9yn/Screenshot-1.png
config stream:
https://i.ibb.co/vd6NMBc/Screenshot-2.png
config docker:
https://i.ibb.co/9TvB598/Screenshot-3.png

check port:
https://i.ibb.co/YjrvBJJ/Screenshot-4.png
https://i.ibb.co/F0SVb14/Screenshot-5.png

Thank you!

**Reply by chaptergy** [reply]
Hm, if the port is not open to the outside that probably means there is some kind of firewall blocking the request on your machine?

**Reply by alebupal** [reply]
> Hm, if the port is not open to the outside that probably means there is some kind of firewall blocking the request on your machine?

Tvheadend I have it on a debian and npm on another debian with docker.
Currently I have the port open on the router and I don't have any kind of problem.

**Reply by chaptergy** [reply]
So you have also forwarded that port in your router if necessary? Then go ahead and run `sudo lsof -i -P -n | grep LISTEN` on your host running npm. If you find port `9982` the problem is somewhere outside of npm an docker, and there has to be some other intermediary thing blocking it like a firewall or a router.

**Reply by alebupal** [reply]
In the router I have the port open, but when I do the tests I disable it to avoid problems.
I have run this: sudo lsof -i -P -n | grep :9982
https://i.ibb.co/NNYmVxR/Screenshot-1.png

Thank you!

**Reply by chaptergy** [reply]
Then the issue is not with you npm setup but somewhere else. But I don't really understand your setup. So you have two hosts, one running tvheadend and on running npm? Then there is a _single_ router in front of _both_ hosts? And this router has a port forwarding for port `80` and `443` to the npm host and port `9982` forwarded to the tvheadend host? It that's the case that's the issue. You need to port forward port `9982` to your npm host, not the tvheadend host. All ports except port `81` listed in your docker-compose need to be port forwarded in your router to your npm instance.

**Reply by alebupal** [reply]
ok, I thought that with open port 80 and 443 everything would work 🤦‍♂️. 
I have opened port 9982 for the npm host (192.168.5.5) and it works fine now.

thank you very much for your help!

**Reply by albguru** [reply]
I know this has been a while but can you share the exact configuration for making this work?
